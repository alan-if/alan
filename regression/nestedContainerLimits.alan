The l Isa location
End The l.

The hero Isa actor
    Container
        Limits
            weight 10
                Else "That weighs too much."
End The hero.

The bag Isa object At l
    Container
End The bag.


Syntax take = take (o).

Syntax put_in = put (o) 'in' (c)
    Where c Isa Container
        Else "You can only put things in containers."

Syntax drop = drop (o).


Add To Every object
    Has
        weight 7.

    Verb take
        Does
            Locate o In hero.
            List hero.
            List bag.
    End Verb.

    Verb put_in
        When o
            Does
                Locate o in c.
                List c.
                List hero.
    End Verb.

    Verb drop
        Does
            Locate o here.
            List hero.
            List bag.
            Look.
    End Verb.
    
End Add.

The stone1 Isa object At l.
End The stone1.

The stone2 Isa object At l.
End The stone2.


Start At l.
