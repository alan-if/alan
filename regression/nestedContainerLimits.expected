########## nestedContainerLimits ##########

        No warnings or errors detected.



L
There is a bag, a stone1 and a stone2 here.

> take stone1
The hero carries a stone1. The bag is empty.

> take stone2
That weighs too much.

> put stone1 in bag
The bag contains a stone1. The hero is empty-handed.

> put stone2 in bag
The bag contains a stone2 and a stone1. The hero is empty-handed.

> take bag
That weighs too much.

> drop stone1
The hero is empty-handed. The bag contains a stone2.

L
There is a bag here. The bag contains a stone2. There is a stone1 here.

> drop stone2
The hero is empty-handed. The bag is empty.

L
There is a bag, a stone1 and a stone2 here.

> drop bag
The hero is empty-handed. The bag is empty.

L
There is a bag, a stone1 and a stone2 here.

> take bag
The hero carries a bag. The bag is empty.

> put stone1 in bag
The bag contains a stone1. The hero carries a bag.

> put stone2 in bag
The bag contains a stone2 and a stone1. The hero carries a bag.

> 

Do you want to UNDO, RESTART, RESTORE or QUIT ? 
